patterns/blueprint-vpc-lattice/cluster1/main.tf

```diff
@@ -0,0 +1,193 @@
+  azs             = local.azs
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 4, k)]
+  public_subnets  = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k + 48)]
+
+  enable_nat_gateway = true
+  single_nat_gateway = true
+
+  public_subnet_tags = {
+    "kubernetes.io/role/elb" = 1
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/role/internal-elb" = 1
+  }
+
+  tags = local.tags
+}
+
+################################################################################
+# EKS Addons (demo application)
+################################################################################
+
+module "addons" {
+  source  = "aws-ia/eks-blueprints-addons/aws"
+  version = "~> 1.0"
+
+  cluster_name      = module.eks.cluster_name
+  cluster_endpoint  = module.eks.cluster_endpoint
+  cluster_version   = module.eks.cluster_version
+  oidc_provider_arn = module.eks.oidc_provider_arn
+
+  # EKS Addons
+  eks_addons = {
+    coredns    = {}
+    kube-proxy = {}
+    vpc-cni = {
+      preserve    = true
+      most_recent = true # Must be 1.14.0 or higher
+
+      timeouts = {
+        create = "25m"
+        delete = "10m"
+      }
+
+    }
+  }
+  enable_aws_gateway_api_controller = true
+  aws_gateway_api_controller = {
+    repository_username = data.aws_ecrpublic_authorization_token.token.user_name
+    repository_password = data.aws_ecrpublic_authorization_token.token.password
+    # awsRegion, clusterVpcId, clusterName, awsAccountId are required for case where IMDS is NOT AVAILABLE, e.g Fargate, self-managed clusters with IMDS access blocked
+    set = [{
+      name  = "clusterVpcId"
+      value = module.vpc.vpc_id
+    },
+    {
+      name = "clusterName"
+      value = module.eks.cluster_name
+    },
+    {
+      name = "awsAccountId"
+      value = local.region
+    },
+    {
+      name = "awsAccountId"
+      value = data.aws_caller_identity.identity.account_id
+    },
+    {
+      name = "awsRegion"
+      value = local.region
+    }
+    ]
+
+  }
+  tags = local.tags
+}
+
+data "aws_ec2_managed_prefix_list" "ipv4" {
+  name = "com.amazonaws.${data.aws_region.current.name}.vpc-lattice"
+}
+
+data "aws_ec2_managed_prefix_list" "ipv6" {
+  name = "com.amazonaws.${data.aws_region.current.name}.ipv6.vpc-lattice"
+}
+
+
+# configure security group to receive traffic from the VPC Lattice network. You must set up security groups so that they allow all Pods communicating with VPC Lattice to allow traffic from the VPC Lattice managed prefix lists. Lattice has both an IPv4 and IPv6 prefix lists available
+resource "aws_security_group_rule" "vpc_lattice_ipv4_ingress" {
+  description       = "VPC lattice ipv4 ingress"
+  type              = "ingress"
+  security_group_id = module.eks.cluster_security_group_id
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  prefix_list_ids   = [data.aws_ec2_managed_prefix_list.ipv4.id]
+}
+
+resource "aws_security_group_rule" "vpc_lattice_ipv6_ingress" {
```

Comment:
if the pattern is not utilizing IPv6, do we need this?

Reply:
I have included them assuming that even dual stack and pure IPV6 VPCs could reach them via the lattice services regardless of the service source VPC is ipv4/6 . I have removed it now

Reply:
removed them as expected